kubernetes/template.yaml.jinja

```diff
@@ -78,17 +79,45 @@ spec:
         - "--job_name={{ job }}"
         - "--worker_hosts={{ worker_hosts() }}"
         - "--ps_hosts={{ ps_hosts() }}"
+        volumeMounts:
+{% if credential_secret_name != "" %}
+        - name: credential
+          mountPath: /etc/credential
+          readOnly: true
 {% endif %}
+{% if job == "worker" and i == 0 %}
+        - name: traindir
```

Comment:
The result of training should typically be in distributed storage specified in the `{{ train_dir }}` variable, rather than mounted locally. We'd want to discourage people from writing results locally on worker 0 for distributed training.